src/backend/marsha/core/serializers/video.py

```diff
@@ -367,26 +369,34 @@ def validate(self, attrs):
         # It is named "user" by convention in the `rest_framework_simplejwt` dependency we use.
         user = self.context["request"].user
         video = get_object_or_404(Video, pk=user.id)
+
+        if not attrs.get("email"):
+            raise serializers.ValidationError({"email": "Email is mandatory."})
+
         if video.is_scheduled is False:
             raise serializers.ValidationError(
                 {"video": f"video with id {user.id} doesn't accept registration."}
             )
 
         if not attrs.get("video_id") and isinstance(user, TokenUser):
             attrs["video_id"] = user.id
-            # consumer_site is defined if context_id exists in the token
-            attrs["consumer_site"] = (
-                Playlist.objects.get(
-                    lti_id=user.token.payload["context_id"]
-                ).consumer_site
-                if user.token.payload.get("context_id")
-                else None
+
+            is_lti = (
+                user.token.payload.get("context_id")
+                and user.token.payload.get("consumer_site")
+                and user.token.payload.get("user")
+                and user.token.payload["user"].get("id")
             )
 
-            if user.token.payload.get("user"):
-                attrs["lti_user_id"] = user.token.payload["user"]["id"]
+            if is_lti:
+                attrs["consumer_site"] = get_object_or_404(
+                    ConsumerSite, pk=user.token.payload["consumer_site"]
+                )
```

Comment:
I don't really understand why you are fetching this data in the database ?

Reply:
If I assign directly the UUID I get an error value, ValueError: Cannot assign "'5ac6ec8b-9857-4ad8-..